sympy/calculus/util.py

```diff
@@ -410,6 +415,14 @@ def periodicity(f, symbol, check=False):
     >>> periodicity(sin(4*x)**cos(2*x), x)
     pi
     >>> periodicity(exp(x), x)
+
+    >>> class F(Function):
+    ...     nargs = 1
+    ...     period = sin.period     # sin's period method calls _period method,
+    ...                             # so _period method must be defined as well.
+    ...     _period = sin._period
```

Comment:
Both here and in the tests you can use something simpler like
```
class F(function):
    nargs = 1
    def period(self, sym):
        arg = self.args[0]
        return 1/arg.coeff(sym)
```

Reply:
This certainly looks far more better. Thanks!